stratum/lib/security/credentials_manager.cc

```diff
@@ -67,13 +67,11 @@ ::util::Status CredentialsManager::Initialize(bool secure_only) {
       server_credentials_ = ::grpc::InsecureServerCredentials();
     }
   } else {
-    // Verify that the certificate files exist and can be read
+    // Verify that the certificate files exist and are regular (non-symlink) files
     // If files are not present or not accesible, method will return with nullptr
-    std::ifstream ifile1, ifile2, ifile3;
-    ifile1.open(FLAGS_server_key_file);
-    ifile2.open(FLAGS_server_cert_file);
-    ifile3.open(FLAGS_ca_cert_file);
-    if(ifile1 && ifile2 && ifile3) {
+    if (stratum::hal::VerifyRegularFile(FLAGS_ca_cert_file) == ::util::OkStatus() &&
```

Comment:
Replace `VerifyRegularFile` with an `IsRegularFile` predicate. See notes elsewhere on the rationale.

```
if (IsRegularFile(FLAGS_ca_cert_file) && ...)
```

Reply:
Fixed